src/mscorlib/shared/System/ArgumentException.cs

```diff
@@ -64,13 +64,12 @@ public ArgumentException(String message, String paramName)
         protected ArgumentException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
-            _paramName = info.GetString("ParamName");
+            throw new PlatformNotSupportedException();
         }
 
         public override void GetObjectData(SerializationInfo info, StreamingContext context)
         {
             base.GetObjectData(info, context);
```

Comment:
We discussed it in https://github.com/dotnet/corefx/pull/20220 and landed on this with the idea that it would be more consistent with exceptions that don't override GetObjectData. I agree that neither case is quite idea.